### DISCUSSION TITLE
ActionMax - RetroPie

### DISCUSSION THREAD

**Discussion by rupshall** [DirtBagXon/hypseus_singe_data#5] [General]
Hi,

I am currently running the most recent Hypseus Singe build from source on RetroPie.  I have **successfully** tested all my Daphne roms with Hypseus Singe emulator, thank you so much for all the work you have done!!!

I am now playing around with some Singe roms, all have worked perfectly, including ActionMax, but I am not able to get it to work following the steps listed in "00-singe1/actionmax/README.md".

The directions say to:

"Place the actionmax folder as ~/RetroPie/roms/daphne/actionmax - (Note: no .daphne extension)"

When I do as listed above and link without the .daphne extension I am seeing this in the log file:

`
Parameters:
Executing: /opt/retropie/emulators/hypseus/hypseus.sh "/home/pi/RetroPie/roms/daphne/38ambushalley.daphne"
[version] Hypseus Singe: v2.8.0-RPi
[console] Setting alternate home dir: /opt/retropie/emulators/hypseus
2022-01-11 10:11:05.537 INFO  [2003] [reset_logfile@324] Version v2.8.0-RPi
2022-01-11 10:11:05.537 INFO  [2003] [reset_logfile@328] Command line: /opt/retropie/emulators/hypseus/hypseus.bin singe vldp -retropath -manymouse -script /home/pi/RetroPie/roms/daphne/38ambushalley.daphne/38ambushalley.singe -framefile /home/pi/RetroPie/roms/daphne/38ambushalley.daphne/38ambushalley.txt -homedir /opt/retropie/emulators/hypseus -fullscreen
2022-01-11 10:11:05.537 INFO  [2003] [reset_logfile@329] CPU : ARMv7 || Mem : 3840 megs
2022-01-11 10:11:05.557 INFO  [2003] [reset_logfile@330] OS : Linux 5.10 || Video : BCM2711
2022-01-11 10:11:05.576 INFO  [2003] [reset_logfile@331] RGB2YUV Function: C
2022-01-11 10:11:05.576 INFO  [2003] [reset_logfile@337] Line Blending Function: C
2022-01-11 10:11:05.576 INFO  [2003] [reset_logfile@344] Audio Mixing Function: C
2022-01-11 10:11:05.776 INFO  [2003] [manymouse_init_mice@301] Using ManyMouse for mice input.
2022-01-11 10:11:05.893 INFO  [2003] [manymouse_init_mice@315] Only 1 mouse found.
2022-01-11 10:11:05.893 INFO  [2003] [manymouse_init_mice@328] #0: Logitech Gamepad F310
2022-01-11 10:11:07.826 INFO  [2003] [ldp_vldp::read_frame_conversions@1043] Framefile parse succeeded. Video/Audio directory is: /home/pi/RetroPie/roms/daphne/38ambushalley.daphne/./
2022-01-11 10:11:07.868 INFO  [2003] [printline@96] Starting Singe version 1.50
2022-01-11 10:11:07.874 INFO  [2003] [printline@96] SINGE: Script Error!
2022-01-11 10:11:08.875 INFO  [2003] [printline@96] SINGE: error compiling script: cannot open singe/../actionmax.daphne/emulator.singe: No such file or directory
2022-01-11 10:11:08.875 INFO  [2003] [printline@96] SINGE: Cannot continue, quitting...
(
`

It's failing because it's looking for "actionmax.daphne/emulator.singe", this doesn't exist.   If I do add the .daphne extension to actionmax and then link 38ambushalley.daphne to it, everything works, but the downside is actionmax.daphne shows up in the game list, and it isn't actually a game.

**Comment by DirtBagXon** [comment]
`cannot open singe/../actionmax.daphne/emulator.singe: No such file or directory`

Apologies those instructions may be incorrect <sup>- _(**Edit:** They have now been updated - see below)_</sup>

I'm afraid you do need the `actionmax.daphne` directory to exist, as this is the path dictated in the LUA.

You will then see an entry for `actionmax.daphne` in the ES list by default I'm afraid.

There are means to hide specific entries in the list in the [retropie forum]( https://retropie.org.uk/forum/topic/5067/hide-roms-in-emulation-station/8). Although, some experimentation will be required.

ActionMax, is a one-off game, as it contains multiple games using the same LUA engine files.

I haven't found a perfect solution for this involving symlinks in RetroPie.....

**Reply by rupshall** [reply]
Thank you for the clarification.

I've been testing with the files Singe game files on archive.org and everything has been really very painless.  I found the easiest way is to pull down the archive unzip into roms/daphne and just make a simlink from [gamename].daphne to [gamename].  I'm not having to modify anything in the singe files to make it work in RetroPie.  It works quite well.

For the Action Max games just this requires unzipping actionmax in the daphne folder and creating 6 simlinks:

ln -s actionmax actionmax.daphne
ln -s actionmax 38ambushalley.daphne
ln -s actionmax bluethunder.daphne
ln -s actionmax hydrosub2021.daphne
ln -s actionmax popsghostly.daphne
ln -s actionmax sonicfury.daphne

Then a reboot or restart of emulation station and everything fires right up.  It's really working quite well, thank you again for all your work on this.  The only issue in as noted above, the side effect of having actionmax showing in the Daphne gamelist.

I'll take a look at the options for hiding entries.

**Reply by DirtBagXon** [reply]
It's a little hacky, but there is another solution....

Pick one of the game names: `38ambushalley`, `bluethunder`, `hydrosub2021`, `popsghostly` or `sonicfury`

Then within the the `roms/daphne/actionmax` directory run the following `perl` _search/replace_ commands:

    perl -p -i -e 's#singe/actionmax/#singe/38ambushalley/#'g *.singe
    perl -p -i -e 's#singe/actionmax/#singe/38ambushalley/#'g */*.singe

Using '_38ambushalley_' as the example here, but any will do.

The second command may fail, depending on your file layout.

This will replace entries within the LUA like this:

    spriteLoad("singe/actionmax/sprite_lighton.png")

    spriteLoad("singe/38ambushalley/sprite_lighton.png")

Then you can remove the `actionmax.daphne` symlink and it should work.

Just make sure you have a backup of the `.singe` files before you run the commands.

It makes the `ActionMax` files a little implementation specific, but should work......

**Reply by rupshall** [reply]
Thank you for that suggestion.  It is a little more hacky as you noted, but the end result is what I was looking for. 

I tested with a fresh unzip with the actionmax file from archive.org.  It only required these commands to work after unzipping with a default RetroPie setup:

```
ln -s actionmax 38ambushalley.daphne
ln -s actionmax bluethunder.daphne
ln -s actionmax hydrosub2021.daphne
ln -s actionmax popsghostly.daphne
ln -s actionmax sonicfury.daphne
```

`perl -p -i -e 's#singe/actionmax/#singe/38ambushalley/#'g actionmax/*.singe
`

**Comment by DirtBagXon** [comment]
If you are experiencing issues playing **ActionMax** on _SBC's_ (_like the Raspberry Pi_) see here:

### ActionMax Pi

https://github.com/DirtBagXon/actionmax-pi

**Comment by DirtBagXon** [comment]
### Cleaner "-retropath" Setup

The `actionmax` game code has been altered to allow users to alter the standard **MYDIR** variable for `actionmax` games:

```
gameID          = "38AmbushAlley"
lengthIntro     = 354
lengthGame      = 27819
lengthMenu      = 96
backgroundFrame = 819
highThreshold   = 127
lowThreshold    = 20
sensorTop       = 182
sensorLeft      = 288
sensorX         = 312
sensorY         = 205

MYDIR = "singe/actionmax"

```

Also if the engine supports the _LUA lfs_ plugin, it should auto detect a `-retropath` environment and change **MYDIR** automatically.

`hypseus` supports the auto detection, the older `actionmax` emulator _does not_ and will require a manual setting of **MYDIR**

In either case, the `perl` commands above and manual alteration of the main `emulator.singe` file should no longer be needed.

See: https://github.com/DirtBagXon/hypseus_singe_data/blob/master/00-singe1/actionmax/README.md
